models/enterprise_sonic/dhcp_relay/deleted_example_02.txt

```diff
@@ -0,0 +1,68 @@
+# Using deleted
+#
+# Before State:
+# -------------
+#
+# sonic# show running-configuration interface
+# !
+# interface Eth1/1
+#  mtu 9100
+#  speed 400000
+#  fec RS
+#  no shutdown
+#  ip address 81.1.1.1/24
+#  ip dhcp-relay 91.1.1.1 92.1.1.1 vrf VrfReg1
+#  ip dhcp-relay max-hop-count 5
+#  ip dhcp-relay vrf-select
+#  ip dhcp-relay policy-action append
+#  ipv6 address 81::1/24
+#  ipv6 dhcp-relay 91::1 92::1
+#  ipv6 dhcp-relay max-hop-count 5
+# !
+# interface Eth1/2
+#  mtu 9100
+#  speed 400000
+#  fec RS
+#  no shutdown
+#  ip address 61.1.1.1/24
+#  ip dhcp-relay 71.1.1.1 72.1.1.1 73.1.1.1
+#  ip dhcp-relay source-interface Vlan100
+#  ip dhcp-relay link-select
+#  ip dhcp-relay circuit-id %h:%p
+# !
+
+  - name: Delete all DHCP relay configurations for interface Eth1/1
+    dellemc.enterprise_sonic.sonic_dhcp_relay:
+      config:
+        - name: 'Eth1/1'
+          ipv4:
```

Comment:
The presence of this line, with nothing after it, should be sufficient for deletion of all ipv4 dhcp_relay configuration on this interface. (If the list of server addresses is explicitly deleted, as shown here, other attributes would remain present after processing the deletion. Don't worry about that case in the deletion examples, though. I am just requesting that the following two lines be deleted in this example.)

Reply:
We had a meeting today to discuss the implementation details and problem that you posted via email and have done more research on existing precedents. I understand that your proposed implementation will work for doing bulk deletion of the defined servers and the way that I requested will not work for this model. For now, it is okay to keep this part of the implementation as you have proposed here. I will mark this issue as "Resolved".

*********************************************

Note:

For possible future consideration: There is at least one existing case in the current code that may provide a way to slightly simplify the user interface for bulk deletion: 

l2_interfaces: Deletion of all "trunk" vlans [or the "access" vlan] by specifying a key with no following value.

https://github.com/ansible-collections/dellemc.enterprise_sonic/blob/main/plugins/module_utils/network/sonic/config/l2_interfaces/l2_interfaces.py  (lines 346-361):

```
                    # check for access or trunk is mentioned without value
                    else:
                        # access mentioned wothout value
                        if ('access' in keys) and conf.get('access', None) is None:
                            # get the existing values and delete it
                            if matched.get('access'):
                                conf['access'] = matched.get('access').copy()
                                request = self.get_access_delete_switchport_request(conf, matched)
                                if request:
                                    requests.append(request)
                        # trunk mentioned wothout value
                        if ('trunk' in keys) and conf.get('trunk', None) is None:
                            # get the existing values and delete it
                            if matched.get('trunk'):
                                conf['trunk'] = matched.get('trunk').copy()
                                requests.extend(self.get_trunk_delete_switchport_request(conf, matched))
```

We would like to see if the code shown above works as intended and, if so, why the behavior differs from what is being seen with the current DHCP relay model. But I will take this as an action item and will look into as time permits. We don't want to delay the DHCP Relay implementation while this issue is being investigates, so please proceed with implementation of this model and resource module as you have currently proposed it.